lib/using_restricted_array.rb

```diff
@@ -6,56 +6,105 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Time complexity is O(1) because the arry has a fixed length of 20
+# Space complexity: I think also O(1) since it is a fixed amount that it can hold
 def length(array)
-  raise NotImplementedError
+  count = 0
+
+  until array[count].nil?
+    count +=1
+  end
+  return count
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: if array length does not got past 20, o(1), but if it increases o(n) because the more/less values the more/less time to process
+# Space complexity: pretty much the same as above, there are no new values added past 20 or new arrays created
 def print_array(array)
```

Comment:
⚠️ You are not printing each element and if you test this method you won't get a nice printout of the array contents. 

You need a loop like length.